EventFiltering/PWGHF/HFFilter.cxx

```diff
@@ -1113,6 +1186,9 @@ struct HfFilter { // Main struct for HF triggers
               auto massCandB = RecoDecay::m(std::array{pVec3Prong, pVecFourth}, std::array{massCharmHypos[iHypo], massPi});
               if (std::abs(massCandB - massBeautyHypos[iHypo]) <= deltaMassHypos[iHypo]) {
                 keepEvent[kBeauty4P] = true;
+                if (applyOptimisation) {
+                  dcaXYCharm[iHypo+1] = track.dcaXY();
+                }
```

Comment:
```suggestion
                if (applyOptimisation) {
                  optimisationTree2Prong(collision.globalIndex(), myscores[iHypo-1][0], scores[iHypo-1][1], scores[iHypo-1][2], track.dcaXY());
                }
```

Reply:
@fgrosa , I don't understand why we should put `iHypo - 1` instead of `iHypo`. 
In a general manner, I am a bit confused with the `++iteratorIndex` (because I am more used to `iteratorIndex++` I guess ^^). 
For instance, in the the loop over charm particles, with index `iCharmPart`, we use `++iCharmPart`. As it has been initialized to `0`, its value should be `1` for the first iteration. Here, we write `myscores[iCharmPart][0] = scores[0]` (while I would have gone for  `myscores[iCharmPart - 1][0] = scores[0]` so that we would have `myscores[0][0 = scores[0]` for the first iteration).
Now, in the loop over beauty particles, the index is `iHypo` (also initialized at `0` and incremented via `++iHypo`). There, we fille the optimisation tree with `myscores[iHypo - 1][0]`. 
I think I am missing something because I don't see why we put `iCharmPart` in one case and `iHypo - 1` in the other :)

Reply:
That's a typo from my side actually, it should simply be `myscores[iHypo]`, sorry for the confusion! :)

Reply:
No problem. Actually, it made me realized that I understood the `++i` vs `i++` thing wrong ^^ For some reason I thought `++i` (`i++`) incremented `i` at the beginning (end) of the loop but it is not the case. I did some tests and it gives the same iteration values for `i` ^^

Reply:
Actually, I believe that my initial belief is true when we don't manipulate a `for` loop iterator :)

Reply:
> Actually, I believe that my initial belief is true when we don't manipulate a `for` loop iterator :)

indeed ;)